nixos/modules/services/games/teeworlds.nix

```diff
@@ -1,96 +1,116 @@
 in
 {
-  options = {
-    services.teeworlds = {
-      enable = mkEnableOption (lib.mdDoc "Teeworlds Server");
-
-      openPorts = mkOption {
-        type = types.bool;
-        default = false;
-        description = lib.mdDoc "Whether to open firewall ports for Teeworlds";
-      };
-
-      name = mkOption {
-        type = types.nullOr types.str;
-        default = null;
-        description = lib.mdDoc ''
-          Name of the server. Defaults to 'unnamed server'.
-        '';
-      };
-
-      register = mkOption {
-        type = types.bool;
-        example = true;
-        default = false;
-        description = lib.mdDoc ''
-          Whether the server registers as public server in the global server list. This is disabled by default because of privacy.
-        '';
-      };
-
-      motd = mkOption {
-        type = types.nullOr types.str;
-        default = null;
-        description = lib.mdDoc ''
-          Set the server message of the day text.
-        '';
-      };
-
-      password = mkOption {
-        type = types.nullOr types.str;
-        default = null;
-        description = lib.mdDoc ''
-          Password to connect to the server.
-        '';
-      };
-
-      rconPassword = mkOption {
-        type = types.nullOr types.str;
-        default = null;
-        description = lib.mdDoc ''
-          Password to access the remote console. If not set, a randomly generated one is displayed in the server log.
-        '';
-      };
-
-      port = mkOption {
-        type = types.int;
-        default = 8303;
-        description = lib.mdDoc ''
-          Port the server will listen on.
-        '';
-      };
+  imports = [
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "name" ]
+      [ "services" "teeworlds" "settings" "sv_name" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "register" ]
+      [ "services" "teeworlds" "settings" "sv_register" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "motd" ]
+      [ "services" "teeworlds" "settings" "sv_motd" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "password" ]
+      [ "services" "teeworlds" "settings" "password" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "rcon_password" ]
+      [ "services" "teeworlds" "settings" "sv_rcon_password" ])
+    (mkRenamedOptionModule
+      [ "services" "teeworlds" "port" ]
+      [ "services" "teeworlds" "settings" "sv_port" ])
+    (mkRemovedOptionModule [ "services" "teeworlds" "extraOptions" ]
+      "Use services.teeworlds.settings instead.")
+  ];
+
+  options.services.teeworlds = {
+    enable = mkEnableOption (lib.mdDoc "Teeworlds Server");
+
+    openPorts = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to open firewall ports for Teeworlds
+      '';
+    };
 
-      extraOptions = mkOption {
```

Comment:
We should keep this for backwards compatibility and append those lines to the generated config. Maybe add a warning to switch to `settings` if `extraOptions` is defined.